### PR TITLE
Test case for check iam_root_mfa_enabled

### PR DESCRIPTION
**Pull request by Gunjan-Katre-Comprinno**
Context
This PR adds unit tests for the iam_root_mfa_enabled check, which ensures that AWS root account has Multi-Factor Authentication (MFA) enabled. Enforcing MFA for the root user is a critical security best practice to prevent unauthorized access.

Description
The test suite covers the following scenarios:

- Root account with MFA enabled – Should pass the check.
- Root account without MFA enabled – Should fail the check.
- Credential report stuck in generation (timeout) – Should return an unknown status.
- Empty credential report returned – Should return an unknown status.
- Credential report missing the mfa_active column – Treated as MFA not enabled, should fail.
- ClientError during report generation (e.g., AccessDenied) – Should return an unknown status with error details.

Mocks are used to simulate AWS IAM client behavior, and test coverage ensures all logical branches of the check are validated.

Checklist

- New test cases added
- Mocks implemented for deterministic test behavior
- All test cases validate check logic and edge cases
- Code follows formatting and docstring conventions

License
I confirm that this contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: library/aws/tests/iam/test_iam_root_mfa_enabled.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            error_response={"Error": {"Code": "AccessDenied", "Message": "Denied"}},
+            operation_name="GenerateCredentialReport"
+        )
+
+        report = self.check.execute(self.mock_session)
+
+        assert report.status == CheckStatus.UNKNOWN
+        assert len(report.resource_ids_status) == 1
+        assert report.resource_ids_status[0].status == CheckStatus.UNKNOWN
+        assert "Denied" in (report.resource_ids_status[0].exception or "")
```
Comment (aafaq-rashid-comprinno) [inline library/aws/tests/iam/test_iam_root_mfa_enabled.py:123]: The test suite for `iam_root_mfa_enabled` is comprehensive and mirrors the quality of `iam_password_policy_lowercase`, covering all key outcomes (enabled, disabled, missing field, timeout, client error), with well-structured mocks, clear assertions on `status` and `summary`, and robust error handling.

**Review by aafaq-rashid-comprinno — APPROVED** [review]
